### ISSUE TITLE
Flaky tests with httpbin.org

### ISSUE THREAD

**Issue by ml31415** [geventhttpclient/geventhttpclient#198]
Some of our tests rely on connecting to external hosts, mainly `httpbin.org`. While this is a great little tool, it currently causes quite some flaky tests. Shall we start to self-host `httpbin`? Or find some offline solution?

**Comment by ml31415** [comment]
```
______________________________ test_multi_queries_greenlet_safe ______________________________

    @pytest.mark.network
    def test_multi_queries_greenlet_safe():
        client = httpbin_client(concurrency=3)
        group = gevent.pool.Group()
        event = gevent.event.Event()
    
        def run(i):
            event.wait()
            response = client.get("/")
            return response, response.read()
    
        count = 0
        ok_count = 0
    
        gevent.spawn_later(0.2, event.set)
        for response, content in group.imap_unordered(run, range(5)):
            # occasionally httpbin.org will return 504 :-/
>           assert response.status_code in [200, 504]
E           AssertionError: assert 502 in [200, 504]
E            +  where 502 = <HTTPSocketPoolResponse status=502 headers={'server': 'awselb/2.0', 'date': 'Tue, 09 Apr 2024 21:01:59 GMT', 'content-type': 'text/html', 'content-length': '122', 'connection': 'keep-alive'}>.status_code
```

This is the flaky test. It might be, that httpbin recently changed their hosting setup and now there are 502 errors instead of 504. I'll provide a quick fix for that. Nevertheless, we should replace these tests.

**Comment by ml31415** [comment]
Just had a look at `pytest-httpbin`. It depends on `httpbin`, which unfortunately comes with a whole zoo of dependencies. Total overkill for what we want to do I guess. If `httpbin` goes on some stricter diet (as attempted here for example: https://github.com/psf/httpbin/issues/26 ), we could consider it again. For now we'd be better of by providing our own server or rewriting the tests.

**Comment by ml31415** [comment]
Now other tests with httpbin started to fail occasionally. Looks like there is no way around some own instance.

**Comment by ml31415** [comment]
Fixed in 8be6f4abc74dbb92850d038b46f89d1bd06eda59
If httpbingo.org should go down for whatever reasons, here are instructions for self hosting it:
https://github.com/mccutchen/go-httpbin/issues/136
